tests/components/nextcloud/test_init.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for the Nextcloud init."""
+
+from unittest.mock import patch
+
+from nextcloudmonitor import (
+    NextcloudMonitorAuthorizationError,
+    NextcloudMonitorConnectionError,
+    NextcloudMonitorError,
+    NextcloudMonitorRequestError,
+)
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.nextcloud.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_URL, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import init_integration, mock_config_entry
+from .const import MOCKED_ENTRY_ID, NC_DATA, VALID_CONFIG
+
+from tests.common import mock_registry, snapshot_platform
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+@pytest.mark.parametrize(
+    ("platform", "expected_entity_count"),
+    [
+        (Platform.BINARY_SENSOR, 6),
+        (Platform.SENSOR, 80),
+        (Platform.UPDATE, 1),
+    ],
+)
+async def test_async_setup_entry(
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    platform: Platform,
+    expected_entity_count: int,
+) -> None:
+    """Test a successful setup entry."""
+    with patch("homeassistant.components.nextcloud.PLATFORMS", [platform]):
+        entry = await init_integration(hass, VALID_CONFIG, NC_DATA)
+
+    assert entry.state == ConfigEntryState.LOADED
+
+    states = hass.states.async_all()
+    assert len(states) == expected_entity_count
+
+    await snapshot_platform(hass, entity_registry, snapshot, entry.entry_id)
```

Comment:
I'd argue that these should be moved to the respective module. Because when I look at an integration tests and I don't see `test_sensor.py` I make the conclusion that the sensor platform is not tested. But this code does test the sensor platform in a way, so we should make the structure reflect that